### PR TITLE
chore: skip update terms if the prices are greater than before

### LINKED ISSUE #735: Update allocation terms
The update [updateAllocationRequest.UpdateTerms](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/allocation.go#L477) option will now only update terms if the blobber's price is lower than before. Change code around [here](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/allocation.go#L1002)

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: kushthedude <kushthedude@gmail.com>

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR  
  
Fixes #735

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | updateAlloc  |
| blobber  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/allocation.go [hunk 1]
```diff
@@ -1006,6 +1007,11 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 		}
 
 		for i, bd := range alloc.BlobberDetails {
+			if bd.Terms.WritePrice <= blobbers[i].Terms.WritePrice &&
+				bd.Terms.ReadPrice <= blobbers[i].Terms.ReadPrice {
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/allocation.go]: Can we treat read price and write price independently. 
* Decrease write pice if blobber terms decreae write price. 
* Decrease read pice if blobber terms decreae read price.

File: code/go/0chain.net/smartcontract/storagesc/allocation.go [hunk 2]
```diff
@@ -1006,6 +1007,11 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
...
+				logging.Logger.Error("Skipping the Update as new price is greater than before for blobbers")
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/allocation.go]: I don't think this is an error, info log should be fine.

**Review by Sriep — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/allocation.go (hunk 1 above)
Comment (kushthedude) [reply code/go/0chain.net/smartcontract/storagesc/allocation.go]: Yes that should be better, making changes.

**Review by Sriep — APPROVED** [review]

**Pull request merged**